packages/transaction-manager/src/index.ts

```diff
@@ -0,0 +1,207 @@
+import { RPC } from "@ckb-lumos/rpc";
+import { bytes } from "@ckb-lumos/codec";
+import type { CKBComponents } from "@ckb-lumos/rpc/lib/types/api";
+import { createInMemoryPendingTransactionStorage } from "./TransactionStorage";
+import type { PendingCell, TransactionStorageType } from "./TransactionStorage";
+import { filterByLumosQueryOptions } from "@ckb-lumos/ckb-indexer/lib/ckbIndexerFilter";
+import { Promisable } from "./storage";
+import { CKBIndexerQueryOptions } from "@ckb-lumos/ckb-indexer/lib/type";
+import { Indexer } from "@ckb-lumos/ckb-indexer";
+
+type OutputsValidator = CKBComponents.OutputsValidator;
+
+interface TransactionManagerType {
+  sendTransaction(tx: Transaction): Promisable<string>;
+  collector(
+    queryOptions: CKBIndexerQueryOptions,
+    usePendingCells?: boolean
+  ): Promisable<CellCollector>;
+  clearCache(): Promisable<void>;
+}
+
+interface TransactionSender {
+  sendTransaction(
+    tx: Transaction,
+    outputsValidator?: "passthrough" | "default"
+  ): Promisable<string>;
+}
+
+interface LumosCellIndexer {
+  collector(options: CKBIndexerQueryOptions): CellCollector;
+}
+
+type Options = {
+  pollIntervalSeconds?: number;
+  // default to in memory storage
+  txStorage?: TransactionStorageType;
+};
+
+type ServiceProviders = {
+  transactionSender: TransactionSender;
+  cellCollectorProvider: LumosCellIndexer;
+};
+
+type ServiceEndPoint = {
+  rpcUrl: string;
+};
+
+function isServiceEndPoint(
+  providers: ServiceEndPoint | ServiceProviders
+): providers is ServiceEndPoint {
+  return typeof providers === "object" && "rpcUrl" in providers;
+}
+
+export class TransactionManager implements TransactionManagerType {
+  private transactionSender: TransactionSender;
+  private cellCollectorProvider: LumosCellIndexer;
+  private txStorage: TransactionStorageType;
+
+  constructor(payload: {
+    providers: ServiceEndPoint | ServiceProviders;
+    options?: Options;
+  }) {
+    if (isServiceEndPoint(payload.providers)) {
+      const { rpcUrl } = payload.providers;
+
+      this.transactionSender = new RPC(rpcUrl);
+      this.cellCollectorProvider = new Indexer(rpcUrl);
+    } else {
+      const { transactionSender, cellCollectorProvider } = payload.providers;
+
+      this.transactionSender = transactionSender;
+      this.cellCollectorProvider = cellCollectorProvider;
+    }
+
+    this.txStorage =
+      payload.options?.txStorage || createInMemoryPendingTransactionStorage();
+  }
+
+  async clearCache(): Promise<void> {
+    await this.txStorage.setTransactions([]);
+  }
+
+  async sendTransaction(
+    tx: Transaction,
+    validator: OutputsValidator = "passthrough"
+  ): Promise<string> {
+    const txHash = await this.transactionSender.sendTransaction(tx, validator);
+    await this.txStorage.addTransaction({ ...tx, hash: txHash });
+    return txHash;
+  }
+
+  async collector(
+    options: CKBIndexerQueryOptions,
+    usePendingCells = true
+  ): Promise<CellCollector> {
```

Comment:
keep the compatible here with the other `indexer.collector(): CellCollector`

```ts
  collector(
    queryOptions?: QueryOptions,
    options?: { usePendingOutputs?: boolean }
  ): CellCollector;
```

Reply:
updated

Reply:
I find this is not updated, if it is updated, the [`common.transfer`](https://github.com/ckb-js/lumos/pull/503#discussion_r1188022100) will work

Reply:
ok，then I'll pass `txStorage` to `TransactionManagerCollector` class, and do async storage    access when `collect` is called

Reply:
updated for code/test/readme/example